File: src/main/java/org/lvtinger/paladin/relation/repo/RelationRepository.java
package org.lvtinger.paladin.relation.repo;

import org.lvtinger.paladin.relation.api.entity.Relation;
import org.springframework.data.jpa.repository.JpaRepository;
import org.springframework.stereotype.Repository;

import java.util.List;

@Repository
public interface RelationRepository extends JpaRepository<Relation, Long> {
    Relation findByOwnerIdAndTargetId(Long ownerId, Long targetId);

    List<Relation> findByOwnerId(Long ownerId);
}

File: src/main/kotlin/org/risesun/paladin/provider/impl/repository/RelationRepository.kt
package org.risesun.paladin.provider.impl.repository

import org.risesun.paladin.provider.api.entity.Relation
import org.springframework.data.jpa.repository.JpaRepository
import org.springframework.stereotype.Repository

@Repository
interface RelationRepository : JpaRepository<Relation, String> {
    fun findByMember(member: Long): MutableList<Relation>
}